tvb_contrib/tvb/contrib/cosimulation/cosimulator.py

```diff
@@ -83,6 +83,27 @@ class CoSimulator(Simulator):
     _cosim_monitors_noncoupling_indices = []
     _cosim_monitors_coupling_indices = []
 
+    def _configure_synchronization_time(self):
+        """This method will set the synchronization time and number of steps,
+           and certainly longer or equal to the integration time step.
+           Moreover, the synchronization time must be equal or shorter
+           than the minimum delay of all existing connections.
+           Existing connections are considered those with nonzero weights.
+        """
+        # The synchronization time should be at least equal to integrator.dt:
+        self.synchronization_time = numpy.maximum(self.synchronization_time, self.integrator.dt)
+        # Compute the number of synchronization time steps:
+        self.synchronization_n_step = iround(self.synchronization_time / self.integrator.dt)
+        # Check if the synchronization time is smaller than the minimum delay of the connectivity:
+        existing_connections = self.connectivity.weights != 0
+        if numpy.any(existing_connections.size):
```

Comment:
This will only be false if the weights matrix is empty. Perhaps you meant w/o `.size`?

Reply:
and you don't really need a nested if here.  This `any(.)` will always return true for any realistic weights matrix, just collapse into a single if-raise.

Reply:
> and you don't really need a nested if here. This `any(.)` will always return true for any realistic weights matrix, just collapse into a single if-raise.

I would suggest to keep it in cases someone wants to perform a simulation without weights for whatever reason.